tests/test_models.py

```diff
@@ -14,3 +14,29 @@ def test_package_manager_get_by_name(name, expected, app, db, auth_env):
         assert expected == pkg_manager
     else:
         assert expected == pkg_manager.name
+
+
+@pytest.mark.parametrize(
+    "from_state,to_state,expected_result",
+    [
+        [RequestStateMapping.complete, RequestStateMapping.stale, True],
+        [RequestStateMapping.failed, RequestStateMapping.stale, True],
+        [RequestStateMapping.in_progress, RequestStateMapping.complete, True],
+        [RequestStateMapping.in_progress, RequestStateMapping.failed, True],
+        [RequestStateMapping.in_progress, RequestStateMapping.in_progress, True],
+        [RequestStateMapping.in_progress, RequestStateMapping.stale, True],
+        # Invalid transitions
+        [RequestStateMapping.complete, RequestStateMapping.failed, False],
+        [RequestStateMapping.complete, RequestStateMapping.in_progress, False],
+        [RequestStateMapping.failed, RequestStateMapping.complete, False],
+        [RequestStateMapping.failed, RequestStateMapping.in_progress, False],
+        [RequestStateMapping.stale, RequestStateMapping.complete, False],
+        [RequestStateMapping.stale, RequestStateMapping.failed, False],
+        [RequestStateMapping.stale, RequestStateMapping.in_progress, False],
+        [RequestStateMapping.stale, RequestStateMapping.stale, False],
```

Comment:
Do we want to add this invalid transactions: 
```suggestion
        [RequestStateMapping.stale, RequestStateMapping.stale, False],
        [RequestStateMapping.failed, RequestStateMapping.failed, False],
        [RequestStateMapping.complete, RequestStateMapping.complete, False],
```
And I have another question: why do we have a different set of invalid transactions in two tests?